pkg/network/usm/ebpf_gotls.go

```diff
@@ -396,13 +396,15 @@ func (p *GoTLSProgram) registerProcess(binID binaryID, pid pid, mTime syscall.Ti
 }
 
 func (p *GoTLSProgram) unregisterProcess(pid pid) {
-	p.lock.Lock()
-	defer p.lock.Unlock()
-
+	p.lock.RLock()
 	binID, found := p.processes[pid]
+	p.lock.RUnlock()
 	if !found {
 		return
 	}
+
+	p.lock.Lock()
+	defer p.lock.Unlock()
```

Comment:
in a second review, the pattern can be problematic
and we need to add:
```
p.lock.Lock()
defer p.lock.Unlock()
binID, found := p.processes[pid]
if !found {
	return
}
```

Reply:
good catch